stubs/Flask-Cors/flask_cors/core.pyi

```diff
@@ -0,0 +1,41 @@
+from logging import Logger
+from typing import Any, Iterable, Pattern, TypeVar
+
+_T = TypeVar("_T")
+_App = Any  # flask is not part of typeshed
+_Response = Any  # flask is not part of typeshed
+_Options = dict[str, Any]
+_MultiDict = Any  # werkzeug is not part of typeshed
+
+LOG: Logger
+ACL_ORIGIN: str
+ACL_METHODS: str
+ACL_ALLOW_HEADERS: str
+ACL_EXPOSE_HEADERS: str
+ACL_CREDENTIALS: str
+ACL_MAX_AGE: str
+ACL_REQUEST_METHOD: str
+ACL_REQUEST_HEADERS: str
+ALL_METHODS: list[str]
+CONFIG_OPTIONS: list[str]
+FLASK_CORS_EVALUATED: str
+RegexObject = Pattern[str]
+DEFAULT_OPTIONS: _Options
+
+def parse_resources(resources: dict[str, _Options] | list[str] | str) -> list[tuple[str, _Options]]: ...
+def get_regexp_pattern(regexp: str | Pattern[str]) -> str: ...
+def get_cors_origins(options: _Options, request_origin: str | None) -> list[str] | None: ...
+def get_allow_headers(options: _Options, acl_request_headers: str | None) -> str | None: ...
+def get_cors_headers(options: _Options, request_headers: dict[str, Any], request_method: str) -> _MultiDict: ...
+def set_cors_headers(resp: _Response, options: _Options) -> _Response: ...
+def probably_regex(maybe_regex: str | Pattern[str]) -> str: ...
+def re_fix(reg: str) -> str: ...
+def try_match_any(inst: str, patterns: Iterable[str]) -> bool: ...
+def try_match(request_origin: str, maybe_regex: str) -> bool: ...
+def get_cors_options(appInstance: _App | None, *dicts: _Options) -> _Options: ...
+def get_app_kwarg_dict(appInstance: _App | None = ...) -> _Options: ...
+def flexible_str(obj: Any) -> str: ...
+def serialize_option(options_dict: _Options, key: str, upper: bool = ...) -> None: ...
+def ensure_iterable(inst: Iterable[_T]) -> Iterable[_T]: ...
```

Comment:
This accepts non-iterables. Giving it the right generic type might require overloads, but not sure if that's worth it.

Reply:
Not sure how to express this correctly. I guess it' then Any for the argument and return type

Reply:
I believe something like this should work (please verify):

```python
_IterableT = TypeVar("_IterableT", bound=Iterable[Any])

@overload
def ensure_iterable(inst: str) -> list[str]: ...
@overload
def ensure_iterable(inst: _IterableT) -> _IterableT: ...
@overload
def ensure_iterable(inst: _T) -> list[_T]: ...
```

It's possible that mypy complains about "overlapping return types", which need to be `# type: ignore`'d.

Reply:
forgot about overload... thanks